zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/FileInfo.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.zeppelin.notebook;
+
+import java.util.concurrent.ConcurrentMap;
+
+import org.apache.commons.lang.StringUtils;
+
+import com.google.common.collect.Maps;
+
+/**
+ * Represents note file info for storage 
+ *
+ */
+public class FileInfo {
+
+  private ConcurrentMap<String, String> fileName;
+  private boolean forceRename;
+  public static final FileInfo EMPTY = new FileInfo();
+  
+  private FileInfo() {
+    fileName = Maps.newConcurrentMap();
+    forceRename = false;
+  }
+  
+  public static FileInfo createInstance() {
+    return new FileInfo();
+  }
+  
+  public String getFileName(String storageName) {
+    if (!fileName.containsKey(storageName)) {
```

Comment:
Thanks @khalidhuseynov Looking forward your refactoring in `NotebookRepo` component in another PR. 😄  BTW, for the file name conversion, What is your plan for all the NotebookRepo implementation. Because it may affect the zeppelin upgrade, so it is better to it all in one major release. Maybe complete it in 0.8